ambry-router/src/test/java/com.github.ambry.router/MockServer.java

```diff
@@ -53,61 +73,145 @@
    */
   public MockBoundedByteBufferReceive send(Send send)
       throws IOException {
-    RequestOrResponseType type = ((RequestOrResponse) send).getRequestType();
     if (!shouldRespond) {
       return null;
     }
+    ServerErrorCode serverError =
+        hardError != null ? hardError : serverErrors.size() > 0 ? serverErrors.poll() : ServerErrorCode.No_Error;
```

Comment:
Whether deletes use this or not is not of concern to the `MockServer`. As far as the `MockServer` is concerned, if a hard error is set, that will be returned. It is designed to be generic.